common/lib/xmodule/xmodule/modulestore/tests/django_utils.py

```diff
@@ -279,6 +281,51 @@ def setUp(self):
         OverrideFieldData.provider_classes = None
         super(SharedModuleStoreTestCase, self).setUp()
 
+    def reset_test_case(self):
+        """
+        Manually run tearDownClass/setUpClass again.
+
+        This is so that if you have a mostly read-only course that you're just
+        modifying in one test, you can write `self.reset_test_case()` at the
+        end of that test and reset the state of the world for other tests in
+        the class.
+        """
+        self.tearDownClass()
+        self.setUpClass()
+
+    @staticmethod
+    def modifies_courseware(f):
+        """
+        Decorator to place around tests that modify course content.
+
+        For performance reasons, SharedModuleStoreTestCase intentionally does
+        not reset the modulestore between individual tests. However, sometimes
+        you might have a test case where the vast majority of tests treat a
+        course as read-only, but one or two want to modify it. In that case, you
+        can do this:
+
+            class MyTestCase(SharedModuleStoreTestCase):
+                # ...
+                @SharedModuleStoreTestCase.modifies_courseware
+                def test_that_edits_modulestore(self):
+                    do_something()
+
+        This is equivalent to calling `self.reset_test_case()` at the end of
+        your test.
+
+        If you find yourself using this functionality a lot, it might indicate
+        that you should be using ModuleStoreTestCase instead, or that you should
+        break up your tests into different TestCases.
+        """
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            return_val = f(*args, **kwargs)
+            obj = args[0]
+            obj.reset_test_case()
+            return return_val
+
+        return wrapper
+
```

Comment:
@cpennington: I realize you already :+1:'d, but may I get a spot check on this method? I haven't written a decorator in a while. :-P